### DISCUSSION TITLE
SDR Gain Settings

### DISCUSSION THREAD

**Discussion by 5foot8** [jekhokie/raspberry-noaa-v2#405] [Q&A] [answered]
What gain settings does everyone use?  

I have tried low 1.7 to high around 37 and have yet to even receive any viable decodes.  Looking at the images I appear to have what appears to be timing marks on the edges of the decodes but no cloud details.  Don't know if this happens independent of any signal.

I currently have a SMART SDR the standard one in the smart square black case, FM broadcast stop filter and the NOAA saw with duel amplifiers.   This is connected to a v-dipole which is roughly aligned north/south.  My SMARTee SDR hopefully will arrive before the end of the day.  It's really unusual because for all purposes it looks like my radios are deaf, yes, I've two of them and I've tried both. I'd expect even a slight decode at the middle of the transition when the satellite is closest to my location, but it looks just like noise.  The decoder is running on this server: 
 https://cloudlog.ddns.net:443/

I'm just trying to fault-find because it's either antenna or radio related.  The capturing and decoding is running well - I just don't think much signal is arriving to be decoded.

Any suggestions?

**Comment by dom-robinson** [comment, marked as answer]
Hiya - so a few thoughts. Definately worth getting things going with 'just' the SDR first. The filters 'improve' things in some cases, but are in no way vital to get a signal, and can add insertion losses or outright break the signal path - so if you can go without extras at first.

Second here are my own settings:

```noaa_15_schedule: true
noaa_15_sdr_device_id: 0
noaa_15_freq_offset: -0
noaa_15_enable_bias_tee: false
noaa_15_gain: 32.8
noaa_15_sun_min_elevation: 6
noaa_15_sat_min_elevation: 30

noaa_18_schedule: true
noaa_18_sdr_device_id: 0
noaa_18_freq_offset: -0
noaa_18_enable_bias_tee: false
noaa_18_gain: 32.8
noaa_18_sun_min_elevation: 6
noaa_18_sat_min_elevation: 30

noaa_19_schedule: true
noaa_19_sdr_device_id: 0
noaa_19_freq_offset: -0
noaa_19_enable_bias_tee: false
noaa_19_gain: 32.8
noaa_19_sun_min_elevation: 6
noaa_19_sat_min_elevation: 30

meteor_m2_schedule: true
meteor_m2_sdr_device_id: 0
meteor_m2_freq_offset: -0
meteor_m2_enable_bias_tee: false
meteor_m2_gain: 16.4
meteor_m2_sun_min_elevation: 10
meteor_m2_sat_min_elevation: 30
```

I use a Noolec SMARTEE v2 SDR for ref.

**Reply by 5foot8** [reply]
Thank you for the information @dom-robinson  - it now appears that the audio / decoding path is faulty.  I tried to run the FM station testing script and initially only got noise, which would explain the almost complete solid noise profile in the spectrogram.  Having looked up the fault code it presented it showed indications that the SOX installation was compromised.  

I used the code which was listed on the site I found the details on `sudo apt-get install libsox-fmt-mp3` to install a missing element of SOX, on retrying the test script again the audio from my local FM station boomed in loud and clear.  So in conclusion I've proved the antenna and cable are working, I've confirmed the SDR is working.

I had high hopes for even a hint of a decode for this mornings passes but alas it does not look much more than noise again.  I'm sure it will be RPi related just running out of simple options to locate the weak link.  Might have to just ditch it all again and do a fresh rebuild of the full RPi.

**Reply by 5foot8** [reply]
I have now completely rebuilt my RPi 4 with just Raspberry-noaa-v2 installed.  Again the fm sample script would not run without me actually installing ncat and the libsox-fmt-mp3 SOX element.  I have removed all dongles and set it up to record this morning.  Disappointed again - for all intents and purposes it looks like the decoder is only receiving noise, there is no hint of decoding and the spectrogram appears to have a blanket noise.

![NOAA-15-20210618-080938-spectrogram](https://user-images.githubusercontent.com/13472948/122526188-f3e7ac80-d011-11eb-8a78-43b15b9b53fc.png)

Any suggestions as to why the decoder/receiver would be generating just pure white noise, I'd expect a hint of a signal at some point in the pass, even if far from optimal.  I've heard the satellites before from my location using a SDRPlay receiver?

This is a totally new installation on a 4GB RPi 4 with a 1TB SSD main drive - install latest 32bit OS , install R-NOAA-V2, install the SOX element stated and ncat which also appeared to not be present when I tried to run the test FM script.  The antenna and cable are functioning because the FM sample script plays our local FM station really well.   I've set the length of the antenna to 53.4 cm as indicated on the RTL-SDR website article on the same subject.  I'm now using a brand new Noolec SMARTee v2 SDR (currently with the bias-t turned off).  I've now tried three different Noolec receivers and all with the same apparent result.

**Reply by 5foot8** [reply]
I don't still appear to be getting a valid signal.  I've never had any pass that looks like it's remotely decoding anything.  Looking at the spectrogram I can't see anything that remotely looks like the smudges of a signal.  Anyone have an image of a spectrogram from a valid and decoded pass so I can compare?  If my understanding of what the spectrogram is showing is correct it appears to be a broad spectrum of white noise.  It's very frustrating!

**Reply by jp112sdl** [reply]
Hey!

> I have now completely rebuilt my RPi 4

The RPi4 is known for its high HF-interferences on radio frequencies, especially when using an external device (i.e. SSD) on the USB3.0 port.
Intel published a workaround for that: [USB 3.0* Radio Frequency Interference on 2.4 GHz Devices](https://www.intel.es/content/dam/www/public/us/en/documents/white-papers/usb3-frequency-interference-papers.pdf)

There are some other discussions regarding interferences [caused by its WiFi](https://www.reddit.com/r/raspberry_pi/comments/gp3jhm/raspberry_pi_4s_usb_30_ports_are_causing/), or HDMI controllers.

Another source for disturbed reception can be the power supply.

So in fact, I was struggling with that, having the same bad images.
<img src="https://user-images.githubusercontent.com/18190139/124222067-48eeec80-db01-11eb-977e-eb99bd3715ca.jpg" height=300 />


I couldn't even get local radio stations tuned in!

My RPi 4 setup now:
- high quality power supply
- disabled WiFi+BT
- nothing connected to the USB3.0 port
- SDR connected to an USB2.0 port via an extension cable (50cm) far away from the RPi 4

Now I am satisfied.

<img src="https://user-images.githubusercontent.com/18190139/124222088-52785480-db01-11eb-8f98-3d09dadce04b.jpg" height=300 />

**Comment by stevecrow74** [comment]
I was thinking the same.

would it be possible to write a script and run it during a noaa low elevation pass, and sweep through the gain to find out the best signal to noise ratio?! 
that was a handy part with running SDR# on the pc. But I’ve noticed the Pi needs different gain settings than the pc even though it’s on the same rig.

**Reply by 5foot8** [reply]
It would be great to have some method for proving / improving the setup. or even a script that says there is no signal present.  I'm convinced my antenna is appropriate and in good condition, finding it hard to locate why I'm not getting any noticeable decode.  I don't know linux code and systems well enough to diagnose where the error lies.  There are plenty of you out there that are obviously getting usable decodes from this system.  I have a few more ideas to work through.  You'll all be the first to know if I get to a usable outcome!
